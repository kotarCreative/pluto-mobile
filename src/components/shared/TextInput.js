/**
 * Standard input component
 *
 * @format
 * @flow
 */

import React, { PureComponent } from 'react'
import { StyleSheet, TextInput as DefaultInput, TouchableHighlight } from 'react-native'
import COLOR from '@/constants/colors'

// styles
const styles = StyleSheet.create({
  input: {
    color: COLOR.WHITE,
    padding: 5,
    textAlign: 'center',
    fontSize: 18
  }
})

export class TextInput extends PureComponent {
  render() {
    return (
      <DefaultInput
        {...this.props}
        placeholderTextColor={COLOR.OFF_WHITE}
        editable={true}
        selectionColor={COLOR.OFF_WHITE}
        style={styles.input} />
    )
  }
}
