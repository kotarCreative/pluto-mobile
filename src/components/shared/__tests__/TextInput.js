// components/__tests__/Hello.tsx
import React from 'react'
import renderer from 'react-test-renderer'

import { TextInput } from '../TextInput'

it('renders correctly with defaults', () => {
  const input = renderer
    .create(<TextInput placeholder="test" />)
    .toJSON()
  expect(input).toMatchSnapshot()
});
