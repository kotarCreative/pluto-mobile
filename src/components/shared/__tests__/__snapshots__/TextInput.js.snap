// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<TextInput
  allowFontScaling={true}
  editable={true}
  placeholder="test"
  placeholderTextColor="#F5FCFF"
  selectionColor="#F5FCFF"
  style={
    Object {
      "color": "#FFFFFF",
      "fontSize": 18,
      "padding": 5,
      "textAlign": "center",
    }
  }
  underlineColorAndroid="transparent"
/>
`;
